src/protocol/modulus_conversion/convert_shares.rs

```diff
@@ -90,6 +90,31 @@ impl ConvertShares {
     }
 }
 
+#[allow(clippy::module_name_repetitions)]
+/// For a given vector of input shares, this returns a vector of modulus converted replicated shares of
+/// `bit_index` of each input.
+pub async fn convert_shares_for_a_bit<F: Field>(
+    ctx: &ProtocolContext<'_, Replicated<F>, F>,
+    input: &[(u64, u64)],
+    num_bits: u8,
+    bit_index: u8,
+) -> Result<Vec<Replicated<F>>, BoxError> {
+    let converted_shares = try_join_all(zip(repeat(ctx), input).enumerate().map(
+        |(record_id, (ctx, row))| async move {
+            let record_id = RecordId::from(record_id);
+            ConvertShares::new(XorShares {
+                num_bits,
+                packed_bits_left: row.0,
+                packed_bits_right: row.1,
+            })
+            .execute_one_bit(ctx.bind(record_id), record_id, bit_index)
```

Comment:
One downside of this `bind()` approach, is that calling `bind()` within a protocol like this means we will have to call `bind()` again in the next protocol. We can't re-use the contexts.

So after we convert these shares, if the next step is to `sort()`, we will need to call `bind()` once per row, once again.

I can live with this for now, but we should think about how we can potentially avoid this (or if this is a problem we are concerned with. `bind()` is a pretty cheap operation...